akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -181,19 +181,30 @@ import akka.stream.scaladsl.Source
             .map(_ => Done)
       }
 
-      composedSource
+      composedSource.via(RunningProjection.stopHandlerWhenFailed(() => handler.stop()))
+    }
+
+    private def tryStartHandler(): Future[Done] = {
```

Comment:
We can move `tryStartHandler` and `tryStopHandler` to the `HandlerLifecycle` trait and make it final.

Then we can remove some repetition inside the projection implementation and we can remove `RunningProjection.stopHandlerWhenStreamCompletedNormally` because it becomes just a `Future.flatMap`

Reply:
`RunningProjection.stopHandlerWhenFailed[T]` can also move the the `HandlerLifecycle`. When we don't even need the `tryStopHandler`

Reply:
I moved the tryStart and tryStop. That's good. See a2c797a

I think I prefer to keep the other utility methods together there in RunningProjection.